# Cellular basis for cortical network aging in primates

This repository hosts data, code, and links to external resources used for this work.

## Publication DOI

[Preprint Link](https://)

## Alignment of Data & Analysis Across Scales 

<p align="center">
  <img src="figs/overview.png" width="1000">
</p>



## Data & Code   

Cross-species alignment of functional domains 

Cortical similarity networks using Morphometric Inverse Divergence (MIND) 

Computation of network-based properties 

Statistical modeling of age effects

Univariate associations with cell type specific abundances

Canonical Correlation Analysis

Human specific cortical cell type enrichment dataset  

Comparison of macaque and human cell types by their aligned functional networks

Macaque to human comparison of cell type composition 



## External Resources & Repositories

MIND network construction
    Code: https://github.com/isebenius/MIND
    Reference:

Functional network cross-species alignment analyses 
    Code: https://github.com/TingsterX/alignment_macaque-human 
    Reference:
    
3D single-cell transcriptomic atlas of macaque monkey
    [Online Website](https://macaque.digital-brain.cn/spatial-omics)
    Reference: Chen 

Canonical correlation analysis (CCA)
    Code: https://github.com/andersonwinkler/PermCCA
    Reference:

Imputed human cell type distributions
    Code:
    Reference

Surface Based Visualizations
    Code: https://github.com/danjgale/surfplot
    Reference:


Neuro Maps?


## Other Tools

Spatial permutation tests utilized the netneurotools, nibabel, and neuromaps Python libraries. Linear mixed effects models were implemented using the lme package in R. Surfaces were rendered using Connectome Workbench. 
















